### PR TITLE
qemu.tests.block_stream: Add case for reset negative speed during blo…

### PR DESCRIPTION
**Pull request by CongLi**
…ck stream.

Signed-off-by: Cong Li <coli@redhat.com>

id: 1370974

### CODE REVIEW

**Comment by CongLi** [comment]
859 17:17:44 INFO | Keyword 'Invalid parameter type' is found in QMP output '{u'error': {u'class': u'GenericError', u'desc': u"Invalid parameter type for 'speed     ', expected: int"}}'.
7276 17:27:44 INFO | PASS 1-smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.block_stream_negative.simple_test.reset_speed.minus_speed

**Comment by CongLi** [comment]
Thanks @xutian , it makes sense.

Have updated the code, could you please help check it again?

Thanks.

File: qemu/tests/block_stream_negative.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+from qemu.tests import blk_stream
+
+
+class BlockStreamNegative(blk_stream.BlockStream):
+
+    def __init__(self, test, params, env, tag):
+        super(BlockStreamNegative, self).__init__(test, params, env, tag)
+
+    @error.context_aware
+    def query_status(self):
```
Comment (vivianQizhu) [inline qemu/tests/block_stream_negative.py]: where do you use this function?

Comment (CongLi) [inline qemu/tests/block_stream_negative.py]: No. Thanks for your comments.

File: qemu/tests/block_stream_negative.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                "speed": expected_speed}
+        response = str(self.vm.monitor.cmd_qmp("block-job-set-speed", args))
+        if "(core dump)" in response:
+            raise error.TestFail("Qemu core dump when reset speed"
+                                 " to a negative value.")
+        if match_str not in response:
+            raise error.TestFail("Fail to get expected result."
+                                 "%s is expected in %s" %
+                                 (match_str, response))
+        logging.info("Keyword '%s' is found in QMP output '%s'." %
```
Comment (xutian) [inline qemu/tests/block_stream_negative.py]: useless line, please drop it

Comment (vivianQizhu) [inline qemu/tests/block_stream_negative.py]: @xutian , I think this logging can be kept for tester awareness.

**Comment by vivianQizhu** [comment]
(2/3) smp_8.1024m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.myprovider.block_stream_negative.simple_test.reset_speed.minus_speed: PASS (1244.39 s)
 (3/3) smp_8.1024m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.myprovider.block_stream_negative.simple_test.reset_speed.string_speed: PASS (1530.86 s)

**Comment by vivianQizhu** [comment]
Acked-by: Qianqian Zhu qizhu@redhat.com

**Pull request merged**